src/fmgc/src/radionav/NavRadioManager.ts

```diff
@@ -8,19 +8,40 @@ export enum TuningMode {
  * This is a placeholder for the new radio nav tuning logic... coming soon to an A32NX near you
  */
 export class NavRadioManager {
-    tuningMode1: TuningMode = TuningMode.Auto;
+    tuningMode: TuningMode = TuningMode.Auto;
 
-    tuningMode2: TuningMode = TuningMode.Auto;
+    manualTuned: boolean;
 
-    tuningMode3: TuningMode = TuningMode.Auto;
+    rmpTuned: boolean;
 
     constructor(public _parentInstrument: BaseInstrument) {
-        SimVar.SetSimVarValue('L:A32NX_FMGC_RADIONAV_1_TUNING_MODE', 'Enum', TuningMode.Manual);
-        SimVar.SetSimVarValue('L:A32NX_FMGC_RADIONAV_2_TUNING_MODE', 'Enum', TuningMode.Manual);
-        SimVar.SetSimVarValue('L:A32NX_FMGC_RADIONAV_3_TUNING_MODE', 'Enum', TuningMode.Manual);
+        SimVar.SetSimVarValue('L:A32NX_FMGC_RADIONAV_TUNING_MODE', 'Enum', TuningMode.Auto);
     }
 
-    update(_: number): void {
-        // Do nothing
+    public update(deltaTime: number, manualTuned: boolean, rmpTuned: boolean): void {
+        if (this.manualTuned !== manualTuned || this.rmpTuned !== rmpTuned) {
+            // too avoid SetSimVar too often
+            this.manualTuned = manualTuned;
+            this.rmpTuned = rmpTuned;
+
+            if (manualTuned) {
+                this.tuningMode = TuningMode.Manual;
+            } else if (rmpTuned) {
+                this.tuningMode = TuningMode.Remote;
+            } else {
+                if (this.tuningMode === TuningMode.Remote) {
+                    // Happens when NAV push button is pushed back
+                    // It resets all the frequencies (real life behavior)
+                    SimVar.SetSimVarValue('K:ADF_ACTIVE_SET', 'Frequency ADF BCD32', 0);
+                    SimVar.SetSimVarValue('K:ADF2_ACTIVE_SET', 'Frequency ADF BCD32', 0);
+                    SimVar.SetSimVarValue('K:NAV1_RADIO_SET_HZ', 'Hz', 0);
+                    SimVar.SetSimVarValue('K:NAV2_RADIO_SET_HZ', 'Hz', 0);
+                    SimVar.SetSimVarValue('K:NAV3_RADIO_SET_HZ', 'Hz', 0);
+                }
```

Comment:
This logic all belongs in the RMP, which actually does the tuning (unless the onside is failed and the offside is tuning it).